### ISSUE TITLE
[469a] No sound in menu or game [Ubuntu 18.04]

### ISSUE THREAD

**Issue by ut99fan** [OldUnreal/UnrealTournamentPatches#14]
Hello Team

Thank you for putting together this awesome release.

Unfortunately for me when running the release OldUnreal-UTPatch469a-Linux on my Ubuntu 18.04 machine **I get absolutely no sound (no music either) (not in game, not in menu)**. Everything else seems to be working fine - I started the match and there seemed to be no issues. In Options-Preferences-Audio it lists as **Audio Driver** "Cluster 3D Audio".

Tested on Unreal Tournament GOTY version installed via Steam (clean download on 2020-09-23). I put the contents of the package OldUnreal-UTPatch469a-Linux.tar.bz2 on top of the Steam installation overwriting everything there as per instructions. I started the game with: **$ ./ut-bin**

I only got couple of warnings:
> WARNING: using preference directory: /home/ut99fan/.utpg/System/
> WARNING: Using system ini file: /mnt/data/SteamLibrary/steamapps/common/Unreal Tournament/System/UnrealTournament.ini
> WARNING: Using user ini file: /mnt/data/SteamLibrary/steamapps/common/Unreal Tournament/System/User.ini
> 
On exit there were no errors.

Happy to debug this any way I can but I need some hand holding here... Thank you very much in advance.

**Comment by stijn-volckaert** [comment]
Could you check if there are any errors in your log file (/home/ut99fan/.utpg/System/UnrealTournament.log)?

Also, please try switching to ALAudio. You can do this by changing AudioDevice to ALAudio.ALAudioSubsystem in the [Engine.Engine] section of your UnrealTournament.ini file.

**Comment by ut99fan** [comment]
Nothing in the .log file - no errors

![Screenshot from 2020-09-24 04-04-36](https://user-images.githubusercontent.com/71790348/94064064-aee47000-fe1b-11ea-96df-0edb3e4a514c.png)

Ok this is probably my system specific as I just remembered that some applications need manually selecting appropriate sound card while other "remember" to use the **Analog Output - Xonar U7** (before I changed it manually it was set to "Digital Output (S/PDIF - HP Dock Audio). This happens to my system after reboot sometimes (the output is switched for unknown reasons to S/PDIF which was not connected like ever). Interestingly enough I played some videos in Firefox browser, used Audacious to listen to music and played videos via VLC without issues without specifying to use Xonar U7 via analog. **Seems more like Ubuntu-related issue**. I will report back after couple of restarts if the ALAudio engine solves the issue and does not need the control panel manual selection workaround.

Even though **solved please leave the issue open for now**. Thank you for quick follow up.

**Comment by kvbev** [comment]
You may consider installing pulseaudio on ubuntu 18.04. I am surprised but a few places reference that ubuntu 18.04 does not use it by default instead using only ALSA. I believe certain apps require pulseaudio now such as firefox.

here is a webpage on how to do that.

https://linuxhint.com/pulse_audio_sounds_ubuntu/

**Comment by ut99fan** [comment]
@stijn-volckaert 
I actually experienced the issue again (after couple of reboots - soundcard switched to something weird) and tried the .ini tweak with ALAudio.ALAudioSubsystem without any luck - it does not help in this particular issue.

@kvbev 
I actually noticed that my Audacious uses as output plugin "PulseAudio Output" which seems to work around the issue with my soundcards messed up. I don't really remember installing pulseaudio package but it's possible I did.

I suppose it's OK to close the issue now but I will leave it up to you if you want me to test something more or close the issue.
